src/core/document.js

```diff
@@ -578,8 +578,20 @@ class Page {
     return tree;
   }
 
-  getAnnotationsData(intent) {
-    return this._parsedAnnotations.then(function (annotations) {
+  getAnnotationsData(handler, task, intent) {
```

Comment:
I know that this is pre-existing code, but would things become simpler if you change this method to be `async` instead?